_posts/2023-02-22-#25038.md

```diff
@@ -7,18 +7,234 @@ authors: [glozow]
+	feerate of the transaction. However, with the current ancestor set-based selection algorithm and
+	possible package topologies, this is difficult to precisely calculate short of building a block
+	template.
+
+	- We have discussed the difficulty of calculating "miner score" in a [previous review club meeting](/26152).
+	Also see discussion in [#23121](https://github.com/bitcoin/bitcoin/pull/23121) and
+	[#26451](https://github.com/bitcoin/bitcoin/pull/26451).
+
+- Descendants of BIP125 signaling transactions cannot be directly replaced: Any descendant of an
+  in-mempool transaction which signals replaceability is also replaceable because
+it will be evicted if its ancestor is replaced. However, a transaction that only conflicts with the
+descendant will be rejected due to it not signaling BIP125 replaceability *explicitly*.
+
+- Transactions must meet RBF fee rules on their own: Package RBF (see [definition](#Definitions)
+  above) is not currently possible. Package RBF would enable, for example, conflicting LN commitment
+transactions (which tend to have fees that are too similar to replace one another) to replace each
+other utilizing fees from a fee-bumping child.
+
+- Further reading on L2 pinning grievances:
+	* [Bringing a nuke to a knife fight: Transaction introspection to stop RBF pinning](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-May/020458.html)
+	* [RBF Pinning with Counterparties and Competing Interest](https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-April/002639.html)
+	* [Pinning : The Good, The Bad, The Ugly](https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-June/002758.html)
+	* [T-bast's Pinning Attacks Document](https://github.com/t-bast/lightning-docs/blob/master/pinning-attacks.md)
+	* [Eltoo Pinning](https://gist.github.com/instagibbs/60264606e181451e977e439a49f69fe1)
+
+A larger list of RBF limitations was discussed last year on [this mailing list
+post](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-January/019817.html) and
+[gist](https://gist.github.com/glozow/25d9662c52453bd08b4b4b1d3783b9ff). Various proposals to amend or rewrite RBF
+policy were discussed on those threads. The discussions resulted in this v3 proposal to address most issues. Additional
+RBF improvements (which do not conflict with v3) are also being considered.
+
+### V3 Policy
+
+The philosophy behind [v3](https://bitcoinops.org/en/topics/version-3-transaction-relay/) is that RBF limitations stem
+from current mempool policy being too permissive in the possible package topologies it accepts. [Package
+limits](https://github.com/bitcoin/bitcoin/blob/master/doc/policy/mempool-limits.md) exist to limit the worst-case
+computational complexity of mempool operations such as block template assembly and new transaction validation. However,
+current limits still allow huge clusters of thousands of transactions.
```

Comment:
I think it's useful to (re-)specify that we're talking about mempool operations - not packages/replacements.